### ISSUE TITLE
Can't evaluate cells - ValueError: Invalid Signature

### ISSUE THREAD

**Issue by heyarne** [clojupyter/clojupyter#89]
I wanted to give `clojupyter` a spin but after a while it seems like I can't evaluate cells anymore. They seem to evaluate normally and immediately but no output is shown; I run clojupyter via `sudo docker run -v (pwd):/home/clojure/notebooks --rm -p 8888:8888 klausharbo/clojupyter`.

The notebook and the `tsv`s are attached.

```
[E 15:04:15.957 NotebookApp] Exception in callback (<zmq.sugar.socket.Socket object at 0x7fe8f7e41670>, <function null_wrapper at 0x7fe8f7e71cf8>)
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/ioloop.py", line 1073, in start
        handler_func(fd_obj, events)
      File "/usr/local/lib/python2.7/dist-packages/tornado/stack_context.py", line 300, in null_wrapper
        return fn(*args, **kwargs)
      File "/usr/local/lib/python2.7/dist-packages/zmq/eventloop/zmqstream.py", line 456, in _handle_events
        self._handle_recv()
      File "/usr/local/lib/python2.7/dist-packages/zmq/eventloop/zmqstream.py", line 486, in _handle_recv
        self._run_callback(callback, msg)
      File "/usr/local/lib/python2.7/dist-packages/zmq/eventloop/zmqstream.py", line 438, in _run_callback
        callback(*args, **kwargs)
      File "/usr/local/lib/python2.7/dist-packages/tornado/stack_context.py", line 300, in null_wrapper
        return fn(*args, **kwargs)
      File "/usr/local/lib/python2.7/dist-packages/zmq/eventloop/zmqstream.py", line 203, in <lambda>
        self.on_recv(lambda msg: callback(self, msg), copy=copy)
      File "/usr/local/lib/python2.7/dist-packages/notebook/services/kernels/handlers.py", line 314, in _on_zmq_reply
        msg = self.session.deserialize(fed_msg_list)
      File "/usr/local/lib/python2.7/dist-packages/jupyter_client/session.py", line 923, in deserialize
        raise ValueError("Invalid Signature: %r" % signature)
    ValueError: Invalid Signature: 'bc2803ea3143aaee0d396f213020e31389e52df17a0bdd39f926d7261af6875f'
```

[climate-emissions.tar.gz](https://github.com/clojupyter/clojupyter/files/3048528/climate-emissions.tar.gz)

**Comment by klausharbo** [comment]
I can reproduce the problem, so there's definitely an issue, but it seems like it may have to do with the Docker artifact, at least it looks like my v0.2.2 local install does not exhibit the same problems, don't why as yet.  I will investigate further.  If you are on Mac or Linux you may want to try out cloning the repo and then compiling and installing a local copy.

-K.

**Comment by pdenno** [comment]
Likewise my local install doesn't have a problem. I haven't tried docker.

**Comment by klausharbo** [comment]
I have hardly any experience with Docker so if anyone feels like taking a stab figuring what might be wrong, please go right ahead!

**Comment by klausharbo** [comment]
I've built a new Docker image based on [reference Docker images provided by the Jupyter
project](https://github.com/jupyter/docker-stacks), specifically the one called [minimal-notebook](https://github.com/jupyter/docker-stacks/tree/master/minimal-notebook).  My testing indicates that the new image does not exhibit the problems described in this issue.

I spent some time trying to find out what caused the problems in the original image, but did not
manage to identify the issue within the time I was willing to spent on that - creating a new image
based on recommended Jupyter  images seemed like a better way forward.

`README.md` is updated to show how to access the new image.

-K.
